opengovsg/postmangovsg
# fix: Save template draft when choosing to finish later

## Problem

Clicking the "Finish this later" button does not save the campaign, causing the user to lose their template draft.

Closes #622.

## Solution

- Add a ref to hold a callback that is executed when the finish later button is clicked
    - Allow child components to set the ref
- If errors occur when executing the callback, navigation back to `/campaigns` is aborted

## Tests

1. Create a new campaign, populate the subject and body
2. Click "Finish this later"
    - Should make a network call to save the template
    - Should navigate back to campaigns dashboard
3. Select the same campaign, the template draft should be there

## TODO

- Draft saving is only done when the current campaign state complies with backend validation rules - is this ok? 
    - Email / Protected email: `subject` and `body` not null, `body` contains `{{ protectedlink }}` for protected email
    - SMS / Telegram: `body` not null
- Protected email: Message B is not persisted since it's not sent to the backend - `localStorage/sessionStorage` maybe?
